src/main/java/com/flowingcode/vaadin/addons/twincolgrid/TwinColGrid.java

```diff
@@ -557,4 +575,42 @@ public void addRightGridSelectionListener(SelectionListener<Grid<T>, T> listener
 		});
     return this;
   }
+
+	public TwinColGrid<T> withShiftMultiselect() {
+		return withSelectRowOnClick().addShiftMultiselect();
+	}
+
+	private TwinColGrid<T> addShiftMultiselect() {
+		forEachSide(side -> {
+			// Disable checkbox
+			side.grid.addClassName("hide-selector-col");
+
+			side.grid.addItemClickListener(c -> {
+				T clicked = c.getItem();
+				if (c.isShiftKey()) {
+					if (side.getLastClickedItem() != null) {
+						GridListDataView<T> dataView = side.grid.getListDataView();

```

Comment:
Doesn't it work with `side.getItems ()`?

Reply:
No because getItems retrieves the list always in the same order, no matter if the user reordered it by some column. GetListDataView retrieves the list as the user is seeing it now.

Reply:
So, this cannot be backported to Vaadin 14 because of https://github.com/FlowingCode/TwinColGridAddon/issues/66... interesting.

Reply:
FWIW, it looks like in Vaadin 14, one could extend Grid<> with something that would extend DataCommunicator to expose fetchFromProvider, which I think results in a stream of display sorted items.  I don't know if that approach would be forwards compatible to 17... or 23.